src/AzureExtension/DevBox/DevBoxInstance.cs

```diff
@@ -251,24 +318,49 @@ public IAsyncOperation<ComputeSystemStateResult> GetState(string options)
         }).AsAsyncOperation();
     }
 
-    // Unsupported operations
-    public IAsyncOperation<ComputeSystemOperationResult> ApplyConfiguration(string configuration) => throw new NotImplementedException();
+    public IAsyncOperation<ComputeSystemThumbnailResult> GetComputeSystemThumbnailAsync(string options)
+    {
+        return Task.Run(async () =>
+        {
+            StateChanged?.Invoke(this, ComputeSystemState.Running);
+
+            var uri = new Uri(Constants.ThumbnailURI);
+            var storageFile = await StorageFile.GetFileFromApplicationUriAsync(uri);
+            var randomAccessStream = await storageFile.OpenReadAsync();
 
-    public IAsyncOperation<ComputeSystemOperationResult> ApplySnapshot(string options) => throw new NotImplementedException();
+            // Convert the stream to a byte array
+            byte[] bytes = new byte[randomAccessStream.Size];

```

Comment:
visual studio keeps complaining that we should say "var" here instead of byte[]  :( so might as well update it. But also lets wrap the contents of this call in a try/catch for good measure.

Reply:
Changed.